src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -127,15 +136,20 @@ private static void setModules(TestInputConfiguration.Builder testInputConfigBui
         } while (!lines.get(lineNo).startsWith("*/"));
     }
 
-    private static String getFullyQualifiedClassName(String filePath, String moduleName) {
+    private static String getFullyQualifiedClassName(String filePath, String moduleName)
+            throws CheckstyleException {
         String fullyQualifiedClassName;
         if (moduleName.startsWith("com.")) {
             fullyQualifiedClassName = moduleName;
         }
         else {
             final String path = SLASH_PATTERN.matcher(filePath).replaceAll("\\.");
-            final int beginIndex = path.indexOf("com.puppycrawl");
             final int endIndex = path.lastIndexOf(moduleName.toLowerCase(Locale.ROOT));
+            if (endIndex == -1) {
+                throw new CheckstyleException("Unable to resolve module name: " + moduleName
+                + ". Please check for spelling errors or specify fully qualified class name.");
```

Comment:
Yes, this is not checking the classpath. It just intends to instantiate a DefaultConfiguration object using the fully qualified class name to avoid anomalies. Thus, we have 2 cases:
1. If the input file belongs to the same test (Example, InputLineLengthDefault.java is an input for LineLengthCheckTest) we are simply asking the user to specify the module name (LineLength in this case) and we are deriving the fully qualified class name from the input file path. (com/puppycrawl/tools... become com.puppycrawl.tools...)
2. If the input file belongs to a different test, or fully qualified class name of check cannot be derived from its path (Example: LambdaTest uses MemberNameCheck), we require the user to specify the fully qualified class name.

Note: ```new DefaultConfiguration(MemberNameCheck.class.getName())``` is exactly the same as ```new DefaultConfiguration("com.puppycrawl.tools.checkstyle.checks.naming.MemberName")```.

Reply:
> avoid anomalies

I am not sure what anomalies is being suggested. All our classes are uniquely named. I changed `getFullyQualifiedClassName` to take the short name as the FQN and changed the input to be the short name with no package and it passed and each module was using unique parameters that it would cause an exception if it didn't get the right module.

For now the wording seems ok, I just prefer to tell them the FQN is a last resort.

Reply:
Now, I remember why we did that! We wanted it to be consistent to the pre-BDD implementations. Discussion is here https://github.com/checkstyle/checkstyle/pull/10106#discussion_r652456891

Reply:
Ok. This is all due to a limitation of `createChecker`. We need to know if we are dealing with a Checker module or a TreeWalker module to know if the base `Checker` needs to be created as it is implied from the little information given to it. `createChecker` could only do this by getting the class instance given to it.

Normal checkstyle doesn't need to know the FQN as it guesses this from https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java#L183 . To me it seems BDD could make use of this, if it doesn't create some recursive issue, and then pass that to the `DefaultConfiguration`.

This is all outside this PR. I am fine with the wording now.

Reply:
Thanks for the suggestion. I'll explore it.

Reply:
we do not need to be very sophisticated now on support of FQN.
we need to finish migration of Filter, I care about FQN the less now.
We need to finish our project and test DBB in 1-2 year of real code changes as it become proven benefit, we can focus on more robust, sophisticated inlined config support.